olive/model/mixin/composite.py

```diff
@@ -0,0 +1,10 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+class CompositeMixin:
+    def set_composite_parent(self, cp):
+        self.composite_parent = cp
+
+    def get_composite_parent(self):
```

Comment:
Should we maintain the constructor function? Take CompositeMixin example, it use `self.composite_parent`. When it comes to subclass which inherts it like:
```python
Class A(CompositeMixin):
    ...
```

Then following code would fail:
```python
a = A()
a.get_composite_parent()
```
See this.
https://stackoverflow.com/questions/39469841/multiple-mixins-and-properties

Reply:
I prefer not have any `__init__` in mixin since the call of multiple __init__ will be much painful. https://www.reddit.com/r/learnpython/comments/ymet1a/writing_mixin_classes_whats_the_best_practice/